core/server/master/src/main/java/alluxio/master/file/InodeTtlChecker.java

```diff
@@ -131,12 +146,21 @@ public void heartbeat() throws InterruptedException {
                 LOG.error("Unknown ttl action {}", ttlAction);
             }
           } catch (Exception e) {
-            LOG.error("Exception trying to clean up {} for ttl check", inode, e);
+            LOG.error("Exception trying to clean up {} for ttl check. Left retries:{}. {}",
+                path, leftRetries - 1, e);
+            if (inode != null && --leftRetries > 0) {
+              failedInodesToRetryNum.put(inode, leftRetries);
+            }
           }
         }
       }
     }
-    mTtlBuckets.removeBuckets(expiredBuckets);
+    // Put back those failed-to-expire inodes for next round retry.
+    if (!failedInodesToRetryNum.isEmpty()) {
```

Comment:
log sth here?

Reply:
that might be too many logs, let me just log those inodes that got retries attempt exhausted only.